nanofi/examples/transmit_payload.c

```diff
@@ -47,9 +47,7 @@ int main(int argc, char **argv) {
   attribute attribute1;
 
   attribute1.key = "some_key";
-  const char * attr_value = "some value";
-  attribute1.value = (void *)attr_value;
-  attribute1.value_size = strlen(attr_value);
+  attribute1.value = "some value";
```

Comment:
I've expected `attribute` to be an owning container of `key`/`value`. We really need to clarify ownership semantics and enforce them through the API.
I support the opaque struct proposal and would provide copying and optionally adopting setters for the struct.